// react-native-svg/lib/commonjs/elements.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Circle", {
  enumerable: true,
  get: function () {
    return _Circle.default;
  }
});
Object.defineProperty(exports, "ClipPath", {
  enumerable: true,
  get: function () {
    return _ClipPath.default;
  }
});
Object.defineProperty(exports, "Defs", {
  enumerable: true,
  get: function () {
    return _Defs.default;
  }
});
Object.defineProperty(exports, "Ellipse", {
  enumerable: true,
  get: function () {
    return _Ellipse.default;
  }
});
Object.defineProperty(exports, "FeBlend", {
  enumerable: true,
  get: function () {
    return _FeBlend.default;
  }
});
Object.defineProperty(exports, "FeColorMatrix", {
  enumerable: true,
  get: function () {
    return _FeColorMatrix.default;
  }
});
Object.defineProperty(exports, "FeComponentTransfer", {
  enumerable: true,
  get: function () {
    return _FeComponentTransfer.default;
  }
});
Object.defineProperty(exports, "FeComposite", {
  enumerable: true,
  get: function () {
    return _FeComposite.default;
  }
});
Object.defineProperty(exports, "FeConvolveMatrix", {
  enumerable: true,
  get: function () {
    return _FeConvolveMatrix.default;
  }
});
Object.defineProperty(exports, "FeDiffuseLighting", {
  enumerable: true,
  get: function () {
    return _FeDiffuseLighting.default;
  }
});
Object.defineProperty(exports, "FeDisplacementMap", {
  enumerable: true,
  get: function () {
    return _FeDisplacementMap.default;
  }
});
Object.defineProperty(exports, "FeDistantLight", {
  enumerable: true,
  get: function () {
    return _FeDistantLight.default;
  }
});
Object.defineProperty(exports, "FeDropShadow", {
  enumerable: true,
  get: function () {
    return _FeDropShadow.default;
  }
});
Object.defineProperty(exports, "FeFlood", {
  enumerable: true,
  get: function () {
    return _FeFlood.default;
  }
});
Object.defineProperty(exports, "FeFuncA", {
  enumerable: true,
  get: function () {
    return _FeComponentTransferFunction.FeFuncA;
  }
});
Object.defineProperty(exports, "FeFuncB", {
  enumerable: true,
  get: function () {
    return _FeComponentTransferFunction.FeFuncB;
  }
});
Object.defineProperty(exports, "FeFuncG", {
  enumerable: true,
  get: function () {
    return _FeComponentTransferFunction.FeFuncG;
  }
});
Object.defineProperty(exports, "FeFuncR", {
  enumerable: true,
  get: function () {
    return _FeComponentTransferFunction.FeFuncR;
  }
});
Object.defineProperty(exports, "FeGaussianBlur", {
  enumerable: true,
  get: function () {
    return _FeGaussianBlur.default;
  }
});
Object.defineProperty(exports, "FeImage", {
  enumerable: true,
  get: function () {
    return _FeImage.default;
  }
});
Object.defineProperty(exports, "FeMerge", {
  enumerable: true,
  get: function () {
    return _FeMerge.default;
  }
});
Object.defineProperty(exports, "FeMergeNode", {
  enumerable: true,
  get: function () {
    return _FeMergeNode.default;
  }
});
Object.defineProperty(exports, "FeMorphology", {
  enumerable: true,
  get: function () {
    return _FeMorphology.default;
  }
});
Object.defineProperty(exports, "FeOffset", {
  enumerable: true,
  get: function () {
    return _FeOffset.default;
  }
});
Object.defineProperty(exports, "FePointLight", {
  enumerable: true,
  get: function () {
    return _FePointLight.default;
  }
});
Object.defineProperty(exports, "FeSpecularLighting", {
  enumerable: true,
  get: function () {
    return _FeSpecularLighting.default;
  }
});
Object.defineProperty(exports, "FeSpotLight", {
  enumerable: true,
  get: function () {
    return _FeSpotLight.default;
  }
});
Object.defineProperty(exports, "FeTile", {
  enumerable: true,
  get: function () {
    return _FeTile.default;
  }
});
Object.defineProperty(exports, "FeTurbulence", {
  enumerable: true,
  get: function () {
    return _FeTurbulence.default;
  }
});
Object.defineProperty(exports, "Filter", {
  enumerable: true,
  get: function () {
    return _Filter.default;
  }
});
Object.defineProperty(exports, "ForeignObject", {
  enumerable: true,
  get: function () {
    return _ForeignObject.default;
  }
});
Object.defineProperty(exports, "G", {
  enumerable: true,
  get: function () {
    return _G.default;
  }
});
Object.defineProperty(exports, "Image", {
  enumerable: true,
  get: function () {
    return _Image.default;
  }
});
Object.defineProperty(exports, "Line", {
  enumerable: true,
  get: function () {
    return _Line.default;
  }
});
Object.defineProperty(exports, "LinearGradient", {
  enumerable: true,
  get: function () {
    return _LinearGradient.default;
  }
});
Object.defineProperty(exports, "Marker", {
  enumerable: true,
  get: function () {
    return _Marker.default;
  }
});
Object.defineProperty(exports, "Mask", {
  enumerable: true,
  get: function () {
    return _Mask.default;
  }
});
Object.defineProperty(exports, "Path", {
  enumerable: true,
  get: function () {
    return _Path.default;
  }
});
Object.defineProperty(exports, "Pattern", {
  enumerable: true,
  get: function () {
    return _Pattern.default;
  }
});
Object.defineProperty(exports, "Polygon", {
  enumerable: true,
  get: function () {
    return _Polygon.default;
  }
});
Object.defineProperty(exports, "Polyline", {
  enumerable: true,
  get: function () {
    return _Polyline.default;
  }
});
Object.defineProperty(exports, "RadialGradient", {
  enumerable: true,
  get: function () {
    return _RadialGradient.default;
  }
});
Object.defineProperty(exports, "Rect", {
  enumerable: true,
  get: function () {
    return _Rect.default;
  }
});
Object.defineProperty(exports, "Stop", {
  enumerable: true,
  get: function () {
    return _Stop.default;
  }
});
Object.defineProperty(exports, "Svg", {
  enumerable: true,
  get: function () {
    return _Svg.default;
  }
});
Object.defineProperty(exports, "Symbol", {
  enumerable: true,
  get: function () {
    return _Symbol.default;
  }
});
Object.defineProperty(exports, "TSpan", {
  enumerable: true,
  get: function () {
    return _TSpan.default;
  }
});
Object.defineProperty(exports, "Text", {
  enumerable: true,
  get: function () {
    return _Text.default;
  }
});
Object.defineProperty(exports, "TextPath", {
  enumerable: true,
  get: function () {
    return _TextPath.default;
  }
});
Object.defineProperty(exports, "Use", {
  enumerable: true,
  get: function () {
    return _Use.default;
  }
});
exports.default = void 0;
var _Circle = _interopRequireDefault(require("./elements/Circle"));
var _ClipPath = _interopRequireDefault(require("./elements/ClipPath"));
var _Defs = _interopRequireDefault(require("./elements/Defs"));
var _Ellipse = _interopRequireDefault(require("./elements/Ellipse"));
var _ForeignObject = _interopRequireDefault(require("./elements/ForeignObject"));
var _G = _interopRequireDefault(require("./elements/G"));
var _Image = _interopRequireDefault(require("./elements/Image"));
var _Line = _interopRequireDefault(require("./elements/Line"));
var _LinearGradient = _interopRequireDefault(require("./elements/LinearGradient"));
var _Marker = _interopRequireDefault(require("./elements/Marker"));
var _Mask = _interopRequireDefault(require("./elements/Mask"));
var _Path = _interopRequireDefault(require("./elements/Path"));
var _Pattern = _interopRequireDefault(require("./elements/Pattern"));
var _Polygon = _interopRequireDefault(require("./elements/Polygon"));
var _Polyline = _interopRequireDefault(require("./elements/Polyline"));
var _RadialGradient = _interopRequireDefault(require("./elements/RadialGradient"));
var _Rect = _interopRequireDefault(require("./elements/Rect"));
var _Stop = _interopRequireDefault(require("./elements/Stop"));
var _Svg = _interopRequireDefault(require("./elements/Svg"));
var _Symbol = _interopRequireDefault(require("./elements/Symbol"));
var _TSpan = _interopRequireDefault(require("./elements/TSpan"));
var _Text = _interopRequireDefault(require("./elements/Text"));
var _TextPath = _interopRequireDefault(require("./elements/TextPath"));
var _Use = _interopRequireDefault(require("./elements/Use"));
var _FeBlend = _interopRequireDefault(require("./elements/filters/FeBlend"));
var _FeColorMatrix = _interopRequireDefault(require("./elements/filters/FeColorMatrix"));
var _FeComponentTransfer = _interopRequireDefault(require("./elements/filters/FeComponentTransfer"));
var _FeComponentTransferFunction = require("./elements/filters/FeComponentTransferFunction");
var _FeComposite = _interopRequireDefault(require("./elements/filters/FeComposite"));
var _FeConvolveMatrix = _interopRequireDefault(require("./elements/filters/FeConvolveMatrix"));
var _FeDiffuseLighting = _interopRequireDefault(require("./elements/filters/FeDiffuseLighting"));
var _FeDisplacementMap = _interopRequireDefault(require("./elements/filters/FeDisplacementMap"));
var _FeDistantLight = _interopRequireDefault(require("./elements/filters/FeDistantLight"));
var _FeDropShadow = _interopRequireDefault(require("./elements/filters/FeDropShadow"));
var _FeFlood = _interopRequireDefault(require("./elements/filters/FeFlood"));
var _FeGaussianBlur = _interopRequireDefault(require("./elements/filters/FeGaussianBlur"));
var _FeImage = _interopRequireDefault(require("./elements/filters/FeImage"));
var _FeMerge = _interopRequireDefault(require("./elements/filters/FeMerge"));
var _FeMergeNode = _interopRequireDefault(require("./elements/filters/FeMergeNode"));
var _FeMorphology = _interopRequireDefault(require("./elements/filters/FeMorphology"));
var _FeOffset = _interopRequireDefault(require("./elements/filters/FeOffset"));
var _FePointLight = _interopRequireDefault(require("./elements/filters/FePointLight"));
var _FeSpecularLighting = _interopRequireDefault(require("./elements/filters/FeSpecularLighting"));
var _FeSpotLight = _interopRequireDefault(require("./elements/filters/FeSpotLight"));
var _FeTile = _interopRequireDefault(require("./elements/filters/FeTile"));
var _FeTurbulence = _interopRequireDefault(require("./elements/filters/FeTurbulence"));
var _Filter = _interopRequireDefault(require("./elements/filters/Filter"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
var _default = exports.default = _Svg.default;
